src/box/iproto.cc

```diff
@@ -3038,36 +3089,70 @@ iproto_free(void)
 	evio_service_stop(&tx_binary);
 }
 
-int
-iproto_rmean_foreach(void *cb, void *cb_ctx)
+static int
+iproto_thread_rmean_foreach(struct rmean *rmean, void *cb, void *cb_ctx)
 {
-	for (size_t i = 0; i < RMEAN_NET_LAST; i++) {
+	int rc = 0;
+	for (size_t i = 0; i < rmean->stats_n; i++) {
+		int64_t mean = rmean_mean(rmean, i);
+		int64_t total = rmean_total(rmean, i);
+		if (((rmean_cb)cb)(rmean->stats[i].name, mean,
+				   total, cb_ctx) != 0)
+			rc = 1;
+	}
+	return rc;
+}
+
+static int
+iproto_rmean_foreach(const char **strings, size_t rmean_last_name,
+		     ptrdiff_t offset, void *cb, void *cb_ctx)
```

Comment:
Can you get 'strings' and rmean_last_name from the first thread instead? (it should always be available)
Also, please rename 'offset' to 'rmean_offset' and add a comment explaining why you use an offset instead of a pointer.

Reply:
Fixed, also as for me passing rmean_last_name  and strings is more clear.